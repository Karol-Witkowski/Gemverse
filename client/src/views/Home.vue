<template>
  <v-container
    class="mt-16 pt-16"
    fluid
  >
    <v-flex class="xs12 sm12 mx-10">
      <v-card>
        <v-card-title>
          <h3 class="grey--text text--darken-2 mb-2">
            Gemverse - a real-time chat
          </h3>
          <span class="grey--text text--darken-2">
            Gemverse is a chat app that keeps you connected with coworker and
            friends wherever you are. User can create own rooms or join to one already made.
            To get started click the button below.
          </span>
        </v-card-title>
        <v-card-actions v-if="!isAuthorized">
          <v-btn
            class="ml-2 mr-4"
            color="blue lighten-2"
            to="/login"
            outlined
          >
            Sign in
          </v-btn>
          <span class="grey--text text--darken-2">or</span>
          <v-btn
            class="ml-4"
            color="blue lighten-2"
            to="/register"
            outlined
          >
            Sign up
          </v-btn>
        </v-card-actions>
        <v-img
          alt="Group of people chatting via multi-platform app"
          class="ml-auto mt-6"
          max-width="60%"
          min-width="320px"
          src="..\assets\img\homeBanner.png"
        />
      </v-card>
    </v-flex>
  </v-container>
</template>

<script>
import { mapGetters } from 'vuex';

export default {
  name: 'Home',
  computed: {
    ...mapGetters(['isAuthorized']),
  },
};
</script>

<style lang="scss">
h3 {
  word-break: normal;
}

span {
  font-size: 1em;
  text-align: justify;
  text-justify: inter-word;
  word-break: normal;
}
</style>
