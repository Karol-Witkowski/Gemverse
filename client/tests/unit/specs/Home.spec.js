import { createLocalVue, mount } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Home from '@/views/Home.vue';

const localVue = createLocalVue();
const vuetify = new Vuetify();
let wrapper;

afterEach(() => {
  wrapper.destroy();
});

describe('Implementation test for Home.vue - unauthorised', () => {
  beforeEach(() => {
    wrapper = mount(Home, {
      localVue,
      mocks: {
        $store: {
          getters: {
            isAuthorized: false,
          },
        },
        vuetify,
      },
    });
  });

  it('Render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Initializes with correct elements', () => {
    expect(wrapper.findAll('.v-btn').length).toEqual(2);
    expect(wrapper.find('[name=login]').text()).toMatch('sign in');
    expect(wrapper.find('[name=register]').text()).toMatch('sign up');
  });
});

describe('Implementation test for Home.vue - authorised', () => {
  beforeEach(() => {
    wrapper = mount(Home, {
      localVue,
      mocks: {
        $store: {
          getters: {
            isAuthorized: true,
          },
        },
        vuetify,
      },
    });
  });

  it('Initializes with correct elements', () => {
    expect(wrapper.findAll('.v-btn').length).toEqual(1);
    expect(wrapper.find('[name=list]').text()).toMatch('room list');
  });
});
