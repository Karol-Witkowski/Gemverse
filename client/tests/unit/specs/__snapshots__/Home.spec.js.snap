// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Implementation test for Home.vue - - unauthorised Render correctly 1`] = `
<div class="container mt-16 pt-16 container--fluid">
  <div class="flex xs12 mx-6">
    <div class="v-card v-sheet theme--light">
      <div class="v-card__title">
        <p class="grey--text mb-2 text--darken-2 title">
          Gemverse - a real-time chat
        </p>
        <p class="grey--text mb-0 text--darken-1">
          Gemverse is a chat app that keeps you connected with coworker and
          friends wherever you are. User can create own public and
          private rooms or join to one already made.
          To get started click the button below and create a new account.
        </p>
      </div>
      <div class="v-card__actions">
        <router-link class="ml-2 mr-3 v-btn v-btn--outlined v-btn--router theme--light v-size--default blue--text text--lighten-2" name="login"><span class="v-btn__content">
          sign in
        </span></router-link> <span class="grey--text text--darken-2">or</span>
        <router-link class="ml-3 v-btn v-btn--outlined v-btn--router theme--light v-size--default blue--text text--lighten-2" name="register"><span class="v-btn__content">
          sign up
        </span></router-link>
      </div>
      <!---->
      <div aria-label="Group of people chatting via multi-platform app" role="img" class="v-image v-responsive ml-auto mt-6 theme--light" style="min-width: 320px; max-width: 60%;">
        <transition-stub name="fade-transition" mode="in-out">
          <div class="v-image__image v-image__image--preload v-image__image--cover" style="background-position: center center;"></div>
        </transition-stub>
        <div class="v-responsive__content"></div>
      </div>
    </div>
  </div>
</div>
`;
